extensions_admin/pulp_docker/extensions/admin/upload.py

```diff
@@ -30,12 +30,16 @@ def __init__(self, context):
 
     def determine_type_id(self, filename, **kwargs):
         """
-        We only support one content type, so this always returns that.
+        Determine whether it is a V1 or V2 image
 
         :return: ID of the type of file being uploaded
         :rtype:  str
         """
-        return constants.IMAGE_TYPE_ID
+        image_manifest = tarutils.get_image_manifest(filename)
+        if isinstance(image_manifest, list):
+            return constants.IMAGE_TYPE_ID
+        elif image_manifest['schemaVersion']:
```

Comment:
i do not feel comfortable to access the key like this, what about using the get method?